### PR TITLE
wire-compiler: support rpcRole and rpcCallStyle arguments

### PR DESCRIPTION
**Pull request by mgenov**
The wire-compiler now accepts --rpcRole and --rpcCallStyle arguments.

### CODE REVIEW

**Comment by mgenov** [comment]
Tried to add tests in WireCompilerTest to verify that server and client code is generated, but I'm not able to run these tests in IDEA. 

image
 
I suppose it's issue related to gradle's api usage in Intellij IDEA:
api project(':wire-schema')

**Comment by oldergod** [comment]
Could you explain your use case? Why not using the plugin?

**Comment by mgenov** [comment]
The use-case is that protobuf files are organized in a protorepository and build script is generating source files for different languages (go and kotlin). 

The build pipeline is as follow: 
1. proto api changes are reviewed and submitted (in mycompany-protos repository) 
2. github action 
 * generates sources artifacts for Go and Kotklin
 * checkouts mycompany-protos-go and mycompany-protos-kotlin 
 * copies generated sources into go and kotlin repository
 * commits new changes with version tags
3. github action on push (mycompany-protos-kotlin)
 * builds & uploads api artifacts into internal maven repository

The same tasks could be executed and with gradle but it will require folder structure re-design.

**Comment by oldergod** [comment]
Would you mind rebasing and take inspirations from this PR? https://github.com/square/wire/pull/1466/files
Specially around doc, fallback when parsing fails, and tests.

File: wire-library/wire-compiler/src/main/java/com/squareup/wire/WireCompiler.kt [hunk 1]
```diff
@@ -280,6 +293,16 @@ class WireCompiler internal constructor(
           arg.startsWith(SWIFT_OUT_FLAG) -> {
             swiftOut = arg.substring(SWIFT_OUT_FLAG.length)
           }
+  
+          arg.startsWith(RPC_ROLE_FLAG) -> {
+            val value = arg.substring(RPC_ROLE_FLAG.length)
+            rpcRole = RpcRole.valueOf(value.toUpperCase(Locale.getDefault()))
```
[the remark this answers is not in the document]
Comment (JakeWharton) [reply wire-library/wire-compiler/src/main/java/com/squareup/wire/WireCompiler.kt:299]: And you have to here if you want to support turkish users!

**Comment by oldergod** [comment]
Closing. Please reach out if there's still a need for this.

**Pull request closed without merging**